src/haxeLanguageServer/features/OrganizeImportsFeature.hx

```diff
@@ -0,0 +1,210 @@
+		var child:Null<TokenTree> = token.getFirstChild();
+		if (child == null)
+			return Project;
+		var topLevelPack:String;
+		switch (child.tok) {
+			case Kwd(_):
+				topLevelPack = '$child';
+			case Const(CIdent(s)):
+				topLevelPack = s;
+			default:
+				return Project;
+		}
+		if (stdLibPackages.contains(topLevelPack))
+			return StdLib;
+		if (packageName == null)
+			return Project;
+		if (topLevelPack == packageName)
+			return Project;
+		return Library;
+	}
+
+	static function organizeImportGroups(doc:TextDocument, context:Context, importGroups:Map<Int, ImportGroup>):Array<TextEdit> {
+		var edits:Array<TextEdit> = [];
+
+		for (group in importGroups)
+			edits = edits.concat(organizeImportGroup(doc, context, group));
+
+		return edits;
+	}
+
+	static function organizeImportGroup(doc:TextDocument, context:Context, importGroup:ImportGroup):Array<TextEdit> {
+		ArraySort.sort(importGroup.imports, sortImports);
+		var newImports:String = importGroup.imports.map(i -> i.text).join("\n");
+
+		ArraySort.sort(importGroup.usings, sortImports);
+		var newUsings:String = importGroup.usings.map(i -> i.text).join("\n");
+
+		var newText:String = FormatterHelper.formatText(doc, context, newImports + "\n" + newUsings, TokenTreeEntryPoint.TYPE_LEVEL);
+
+		var edits:Array<TextEdit> = [];
+
+		// remove all existing imports/usings from group
+		for (i in importGroup.imports)
+			edits.push(makeImportEdit(doc, i.token));
+
+		for (i in importGroup.usings)
+			edits.push(makeImportEdit(doc, i.token));
+
+		// insert sorted imports/usings at startOffset
+		var importInsertPos:Position = doc.positionAt(importGroup.startOffset);
+		edits.push(WorkspaceEditHelper.insertText(importInsertPos, newText + "\n"));
+
+		return edits;
+	}
+
+	static function makeImportEdit(doc:TextDocument, token:TokenTree):TextEdit {
+		var range:Range = doc.rangeAt2(token.getPos());
+		// TODO move marker to beginning of next line assumes imports are one line each
+		// maybe look at document whitespace and remove all trailing?
+		range.end.line++;
+		range.end.character = 0;
+
+		var nextLineRange:Range = {
+			start: range.end,
+			end: {
+				line: range.end.line + 1,
+				character: 0
+			}
+		};
+		var lineAfter:String = doc.getText(nextLineRange).trim();
+		if (lineAfter.length <= 0)
+			range.end.line++;
+
+		return WorkspaceEditHelper.removeText(range);
+	}
+
+	static function sortImports(a:ImportInfo, b:ImportInfo):Int {
+		if (a.type < b.type)
+			return -1;
+		if (a.type > b.type)
+			return 1;
+		if (a.text < b.text)
+			return -1;
+		if (a.text > b.text)
+			return 1;
+		return 0;
+	}
+
+	static function determineStartPos(token:TokenTree):Int {
+		return token.pos.min;
+	}
+}
+
+typedef ImportGroup = {
+	var id:Int;
+	var startOffset:Int;
+	var imports:Array<ImportInfo>;
+	var usings:Array<ImportInfo>;
+}
+
+typedef ImportInfo = {
+	var token:TokenTree;
+	var text:String;
+	var type:ImportType;
+}
+
+enum abstract ImportType(Int) {
+	var StdLib;
+	var Library;
```

Comment:
I don't really see how you would be able to tell apart "Library" and "Project"?

Reply:
Well, I'm just looking at your `package` declaration and take the first part of it. Everything that start with the same top level package is considered `Project` level. 
It won't work if you are working on a tink library and you include a bunch of other tink libs, but regular projects will have a different toplevel package - at least that is my assumption.

Reply:
Looking promising! Would it be possible to configure it so that it adds empty lines between sections?

Reply:
it should, if you have a `hxformat.json` config instructing formatter to do so, e.g.:

```json
{
  "emptyLines": {
    "importAndUsing": {
      "betweenImportsLevel": "secondLevelPackage",
      "betweenImports": 1
    }
  }
}
```

should result in 
```haxe
import js.Browser;

import js.jquery.Event;
import js.jquery.JQuery;

import formatter.codedata.TokenList;
```